lib/features/family_goal_tracker/cubit/goal_tracker_cubit.dart

```diff
@@ -0,0 +1,119 @@
+import 'dart:convert';
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:givt_app_kids/core/injection/injection.dart';
+import 'package:givt_app_kids/features/family_goal_tracker/model/family_goal.dart';
+import 'package:givt_app_kids/features/family_goal_tracker/repository/goal_tracker_repository.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+part 'goal_tracker_state.dart';
+
+class GoalTrackerCubit extends Cubit<GoalTrackerState> {
+  GoalTrackerCubit(
+    this._goalTrackerRepository,
+  ) : super(const GoalTrackerState());
+  final GoalTrackerRepository _goalTrackerRepository;
+  String dissmissedGoalKey(String childId) {
+    return '$childId-dissmissedGoalKey';
+  }
+
+  Future<void> getGoal(String childId) async {
+    try {
+      final goals = await _goalTrackerRepository.fetchFamilyGoal();
+
+      // No goals ever set
+      if (goals.isEmpty) {
+        return;
+      }
+
+      // Sort goals by date created, latest first
```

Comment:
I like that you added some comments in this file :) Makes it very readable!

Reply:
It was a lot of logic back then, I was struggling to keep up with it otherwise 😄